### ISSUE TITLE
Uncaught TypeError: dir.contains is not a function

### ISSUE THREAD

**Issue by kompot** [t9md/atom-narrow#227]
[Enter steps to reproduce:]

Suddenly started to get this exception when narrowing with search provder.
0.46.0 on Atom 1.18 to blame?

**Atom**: 1.18.0 x64
**Electron**: 1.3.15
**OS**: Mac OS X 10.12.5
**Thrown From**: [narrow](https://github.com/t9md/atom-narrow) package 0.46.0

### Stack Trace

Uncaught TypeError: dir.contains is not a function

```
At /Users/kompot/.atom/packages/narrow/lib/utils.js:286

TypeError: dir.contains is not a function
    at getProjectPaths (/packages/narrow/lib/utils.js:286:15)
    at Search.checkReady (/packages/narrow/lib/provider/search.js:39:23)
    at Search.start (/packages/narrow/lib/provider/provider-base.js:261:45)
    at Function.start (/packages/narrow/lib/provider/provider-base.js:79:46)
    at Object.narrow (/packages/narrow/lib/main.js:174:18)
    at /packages/narrow/lib/main.js:110:51)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:265:35)
    at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:65
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:610:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:401:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -0:15.7.0 narrow:search-current-project (input.hidden-input)
     -0:08.4.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-nord-atom-syntax-accessibility-custom-comment-contrast.theme-nord-atom-syntax.theme-one-dark-ui)
```

### Non-Core Packages

```
80s-pastel-syntax 1.2.0 
ariake-dark-syntax 0.1.3 
atom-beautify 0.30.3 
atom-clock 0.1.10 
atom-import-js 0.9.0 
atom-jest-snippets 1.1.1 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 1.0.0 
atom-material-ui 2.0.4 
atom-monokai 0.10.9 
atom-ternjs 0.18.3 
atom-typescript 11.0.3 
atom-visual-studio-code-ui 1.3.4 
atomatigit 1.5.5 
busy-signal 1.4.3 
chester-atom-syntax 0.2.0 
dark-pastel-eighties-syntax 0.4.0 
duotone-berries-syntax 0.1.1 
duotone-blue-dark-syntax 1.4.0 
duotone-carrots-syntax 0.2.0 
duotone-dark-amethyst-syntax 2.0.4 
duotone-dark-desert-syntax 0.1.3 
duotone-dark-earth-syntax 2.1.0 
duotone-dark-forest-syntax 2.1.0 
duotone-dark-red-syntax 0.3.2 
duotone-dark-sea-syntax 2.1.0 
duotone-dark-sky-syntax 2.1.0 
duotone-dark-space-syntax 2.1.0 
duotone-dark-syntax 2.1.0 
duotone-darkpool-syntax 1.1.0 
duotone-darksky 1.1.5 
duotone-denim-syntax 0.4.3 
duotone-dusk-syntax 0.3.0 
duotone-light-blue-syntax 1.0.1 
duotone-snow 2.2.0 
duotone-stark-sea-syntax 2.1.0-3 
easy-motion-redux 1.1.1 
editorconfig 2.2.2 
emmet 2.4.3 
expand-region 0.3.0 
flatland-dark 0.2.2 
git-diff-details 1.4.0 
git-plus 7.9.3 
git-status 0.3.4 
git-time-machine 1.5.9 
hey-pane 1.0.0 
improved-chester-atom-syntax 0.3.0 
intentions 1.1.2 
jumpy 3.1.3 
language-babel 2.66.0 
language-diff 0.7.0 
language-javascript-jsx 0.3.7 
language-postcss 1.3.1 
language-pug 0.0.21 
last-cursor-position 0.9.2 
linter 2.2.0 
linter-eslint 8.2.1 
linter-flow 5.6.0 
linter-stylelint 3.10.0 
linter-ui-default 1.6.2 
maximize-panes 0.2.0 
merge-conflicts 1.4.5 
minimap 4.28.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-hide 0.3.0 
minimap-highlight-selected 4.6.1 
minimap-linter 2.0.0 
minimap-pigments 0.2.2 
minimap-selection 4.4.0 
minimap-split-diff 0.3.7 
monokai 0.24.0 
narrow 0.46.0 
nord-atom-syntax 0.9.1 
nord-atom-ui 0.11.0 
pastel-dark-theme 0.1.2 
prettier-atom 0.33.0 
rain-syntax 2.0.2 
rain-ui 1.0.1 
react 0.16.2 
restart-atom 0.2.0 
source-preview-pug 0.2.0 
split-diff 1.4.2 
swackets 0.26.0 
termina-dark-syntax 0.2.0 
tester 1.3.1 
tree-view-git-status 1.4.0 
vim-mode-plus 0.93.0 
zentabs 0.8.8 
```

**Comment by kompot** [comment]
Checked on Atom 1.19 beta 1 - same error. Any suggestions?

**Comment by kompot** [comment]
Rolled back to 0.45.1 on Atom 1.18 - works ok.

**Comment by t9md** [comment]
The original issue you reported have strange combination for narrow version and atom version.

Let me clarify
- Atom v1.19.0-beta1 with narrow v0.46.0 should work.
- Atom v1.18.0 with narrow v0.46.0 is not supported combination, see you cannot install v0.46.0 from v1.18.0
- Atom v1.18.0 with narrow v0.45.1 should work as you said and its newest supported version for Atom v1.18.0.

**Comment by t9md** [comment]
When you switch between beta and stable, it's easily have this unwanted combination, and I know it's not good for normal user(since it's not obvious that is issue).

Ideally Atom should separately manage package directly but it's not. 
You can manually pass ATOM_HOME env to keep package separately.
e.g. `ATOM_HOME=~/.atom-beta atom-beta .`

**Comment by kompot** [comment]
Well, everything is understandable since 0.46 is 1.19+ only. But strangely it throws the same exception on 1.19. 

Will close this as this is probably just some env mixture bug. Should be probably be ok when 1.19 is stable and out for all.

**Comment by t9md** [comment]
You don't need to close, I want to clarify what the issue is.

If you can reproduce with `Atom1.19 + narrow0.46`, then it's narrow's bug.
For me I can't reproduce it, so want to know the steps to repro.

**Comment by kompot** [comment]
Not sure I could help here. Was just playing around with 1.19 and all plugins are shared between main and beta versions, as far as I understand. If you have any thoughts on how to debug further - I'm glad to help.

**Comment by t9md** [comment]
I want to know you can reproduce it now with consistently?
Or is that one time occasion which you  no longer able to reproduce.

**Comment by kompot** [comment]
Yes, reproduction is consistent.

Interestingly enough - with narrow@0.45.1 and 1.19 I have another exception constantly
https://github.com/atom/atom/issues/14834

**Comment by t9md** [comment]
I want to focus `Atom1.19 + narrow0.46`  only, since other combination is not supported in v1.19.0.

If you can reproduce with this combination , I want to know the steps to reproduce, I want to fix it.

**Comment by kompot** [comment]
Yes, I have this exception all the time in 1.19.0-beta2 x64 (Mac OS X) and Narrow 0.46.

1. Open project
2. Select some text in a file belonging to the project.
3. Launch `narrow:search-current-project` command
4. Get the exception in 100% of launches.

Happy to assist with any other details.

**Comment by kompot** [comment]
Atom error report from 1.19

**Atom**: 1.19.0-beta2 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.5
**Thrown From**: [narrow](https://github.com/t9md/atom-narrow) package 0.46.0


### Stack Trace

Uncaught TypeError: dir.contains is not a function

```
At /Users/kompot/.atom/packages/narrow/lib/utils.js:286

TypeError: dir.contains is not a function
    at getProjectPaths (/packages/narrow/lib/utils.js:286:15)
    at Search.checkReady (/packages/narrow/lib/provider/search.js:39:23)
    at Search.start (/packages/narrow/lib/provider/provider-base.js:261:45)
    at Function.start (/packages/narrow/lib/provider/provider-base.js:79:46)
    at Object.narrow (/packages/narrow/lib/main.js:174:27)
    at /packages/narrow/lib/main.js:110:51)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom Beta.app/Contents/Resources/app/src/command-registry.js:265:35)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/command-registry.js:3:65
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom Beta.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:610:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom Beta.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:401:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom Beta.app/Contents/Resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/Applications/Atom Beta.app/Contents/Resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -5:09.2.0 core:cancel (input.hidden-input)
     -5:03.1.0 narrow:search-current-project (input.hidden-input)
  2x -5:01.1.0 settings-view:open (input.hidden-input)
     -4:44.8.0 core:copy (input.hidden-input)
     -1:57.2.0 narrow:search-current-project (input.hidden-input)
  2x -1:54 pane:show-next-item (input.hidden-input)
  4x -1:53.1.0 pane:show-previous-item (input.hidden-input)
     -1:51.2.0 core:move-up (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-nord-atom-syntax-accessibility-custom-comment-contrast.theme-nord-atom-syntax.theme-one-dark-ui)
     -1:48.4.0 narrow:search-current-project (input.hidden-input)
  3x -1:41.8.0 core:move-up (input.hidden-input)
 10x -0:58.9.0 core:select-right (input.hidden-input)
  3x -0:57.9.0 core:select-left (input.hidden-input)
     -0:57.2.0 command-palette:toggle (input.hidden-input)
  2x -0:54.3.0 core:move-down (input.hidden-input)
     -0:53.2.0 core:confirm (input.hidden-input)
  2x -0:53.2.0 narrow:search-current-project (input.hidden-input)
```

### Non-Core Packages

```
80s-pastel-syntax 1.2.0 
ariake-dark-syntax 0.1.3 
atom-beautify 0.30.3 
atom-clock 0.1.12 
atom-import-js 0.10.0 
atom-jest-snippets 1.1.1 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 1.0.0 
atom-material-ui 2.0.4 
atom-monokai 0.10.9 
atom-ternjs 0.18.3 
atom-typescript 11.0.5 
atom-visual-studio-code-ui 1.3.4 
atomatigit 1.5.5 
busy-signal 1.4.3 
chester-atom-syntax 0.2.0 
dark-pastel-eighties-syntax 0.4.0 
duotone-berries-syntax 0.1.1 
duotone-blue-dark-syntax 1.4.0 
duotone-carrots-syntax 0.2.0 
duotone-dark-amethyst-syntax 2.0.4 
duotone-dark-desert-syntax 0.1.3 
duotone-dark-earth-syntax 2.1.0 
duotone-dark-forest-syntax 2.1.0 
duotone-dark-red-syntax 0.3.2 
duotone-dark-sea-syntax 2.1.0 
duotone-dark-sky-syntax 2.1.0 
duotone-dark-space-syntax 2.1.0 
duotone-dark-syntax 2.1.0 
duotone-darkpool-syntax 1.1.0 
duotone-darksky 1.1.5 
duotone-denim-syntax 0.4.3 
duotone-dusk-syntax 0.3.0 
duotone-light-blue-syntax 1.0.1 
duotone-snow 2.2.0 
duotone-stark-sea-syntax 2.1.0-3 
easy-motion-redux 1.1.1 
editorconfig 2.2.2 
emmet 2.4.3 
expand-region 0.3.0 
flatland-dark 0.2.2 
git-diff-details 1.4.0 
git-plus 7.9.3 
git-status 0.3.4 
git-time-machine 1.5.9 
hey-pane 1.0.0 
improved-chester-atom-syntax 0.3.0 
intentions 1.1.2 
jumpy 3.1.3 
language-babel 2.66.0 
language-diff 0.7.0 
language-javascript-jsx 0.3.7 
language-postcss 1.3.1 
language-pug 0.0.21 
last-cursor-position 0.9.2 
linter 2.2.0 
linter-eslint 8.2.1 
linter-flow 5.6.0 
linter-stylelint 3.10.0 
linter-ui-default 1.6.2 
maximize-panes 0.2.0 
merge-conflicts 1.4.5 
minimap 4.28.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-hide 0.3.0 
minimap-highlight-selected 4.6.1 
minimap-linter 2.0.0 
minimap-pigments 0.2.2 
minimap-selection 4.4.0 
minimap-split-diff 0.3.7 
monokai 0.24.0 
narrow 0.46.0 
nord-atom-syntax 0.9.1 
nord-atom-ui 0.11.0 
pastel-dark-theme 0.1.2 
prettier-atom 0.35.1 
rain-syntax 2.0.2 
rain-ui 1.0.1 
react 0.16.2 
restart-atom 0.2.0 
source-preview-pug 0.2.0 
split-diff 1.4.3 
swackets 0.26.0 
termina-dark-syntax 0.2.0 
tester 1.4.0 
tree-view-git-status 1.4.0 
vim-mode-plus 0.93.0 
zentabs 0.8.8 
```

**Comment by t9md** [comment]
Wow, was super easy to repro.
I normally don't use `narrow:search-current-project`, so not noticed.
Will fix soon.

**Comment by t9md** [comment]
sorry for this bug, fixed in v0.46.1(just released).
